src/board/board.py

```diff
@@ -46,22 +45,73 @@ def render_board(self):
         self.render_background()
 
         # Render pieces except current one
-        for tetromino in self.board_tetrominos:
-            tetromino.render_tetromino()
+        for square in self.board_tetrominos_squares:
+            square.render_square()
 
         # Render the ghost tetromino
         self.ghost_tetromino.render_tetromino()
 
         # Render current playable tetromino
         self.current_tetromino.render_tetromino()
 
+    def get_filled_indices(self):
+        """
+        Returns the number of lines filled.
+
+        Returns:
+            list (int): The indices filled.
+
+        """
+        filled_indices = []
+        for j in range(self.height):
+            filled = True
+            for i in range(self.width):
+                if self.board_tetrominos_matrix[i][j] == 0:
+                    filled = False
+            if filled:
+                filled_indices.append(j)
+        return filled_indices
+
+    def clear_lines(self, indices):
+        """
+        Takes in a list of indices that are full and removes all the
+        squares in the row.
+
+        Args:
+            indices (list int): The list of filled indices.
+        """
+
+        #  Needs a copy of the list so it doesn't mutate the original list
+        board_tetrominos_squares_copy = self.board_tetrominos_squares[:]
+        for index in indices:
+            for square in self.board_tetrominos_squares:
+                if square.y == index:
+                    board_tetrominos_squares_copy.remove(square)
+
+        self.board_tetrominos_squares = board_tetrominos_squares_copy
+
+    def drop_lines(self, indices):
+        """
+        Drops the lines based on the given indices.
+
+        Args:
+            indices (list int): The list of filled indices.
+        """
+        lines_dropped = 0
```

Comment:
this var isn't needed either,
can you implement `enumerate` in this for loop instead?
https://stackoverflow.com/questions/522563/accessing-the-index-in-for-loops